User cant login if login and ldapLogin is set

**Describe the bug**
If a user has a `login` and `ldapLogin` entry the user can't login and the message appear

[ERROR] org.kitodo.production.services.data.UserService - Login 'xxx' was found more than once! Affected IDs: 303, 303

The reason behind this error message is the code in

https://github.com/kitodo/kitodo-production/blob/master/Kitodo/src/main/java/org/kitodo/production/services/data/UserService.java#L166-L170

There are 2 queries which get merged but identical users are not merged and delivered as separate entries.

**Possible solution(s)**
1) Still use the two queries but filter identical but duplicated users from the list before the result list is returned
2) Merge the two queries to one query

**Workaround**
Remove the entry in `ldapLogin` if not really needed.

**Release**
3.6.0-SNAPSHOT from 03/01/2023

--- a/Kitodo/src/main/java/org/kitodo/production/services/data/UserService.java
+++ b/Kitodo/src/main/java/org/kitodo/production/services/data/UserService.java
@@ -164,8 +164,7 @@ public class UserService extends ClientSearchDatabaseService<User, UserDAO> impl
      *             if no unique user can be found
      */
     public User getByLdapLoginOrLogin(String login) {
-        List<User> users = getByLoginQuery(login, "from User where ldapLogin = :login");
-        users.addAll(getByLoginQuery(login, "from User where login = :login"));
+        List<User> users = getByLoginQuery(login, "from User where ldapLogin = :login or login = :login");
         return uniqueResult(users, login);
     }
 
@@ -175,6 +174,8 @@ public class UserService extends ClientSearchDatabaseService<User, UserDAO> impl
      * @param login
      *            The login.
      * @return The user.
+     * @throws UsernameNotFoundException
+     *             if no unique user can be found
      */
     public User getByLogin(String login) {
         List<User> users = getByLoginQuery(login, "from User where login = :login");

--- a/Kitodo/src/test/java/org/kitodo/production/services/data/UserServiceIT.java
+++ b/Kitodo/src/test/java/org/kitodo/production/services/data/UserServiceIT.java
@@ -247,4 +247,16 @@ public class UserServiceIT {
         assertEquals("Returned authenticated user was wrong", "kowal", authenticatedUser.getLogin());
         SecurityTestUtils.cleanSecurityContext();
     }
+
+    @Test
+    public void returnCorrectUserIndependentOfLoginOrLdapLoginByLogin() {
+        User user = userService.getByLdapLoginOrLogin("verylast");
+        assertEquals("Returned user was wrong", "User, Very last", user.getFullName());
+    }
+
+    @Test
+    public void returnCorrectUserIndependentOfLoginOrLdapLoginByLdapLogin() {
+        User user = userService.getByLdapLoginOrLogin("doraLDP");
+        assertEquals("Returned user was wrong", "Dora, Anna", user.getFullName());
+    }
 }

